src/components/ArticlePage.jsx

```diff
@@ -0,0 +1,41 @@
+import { useParams } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { getArticleById } from './apiCalls';
+import { Link } from 'react-router-dom';
+export default function ArticlePage() {
+    const { article_id } = useParams();
+    const [article, setArticle] = useState(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        setIsLoading(true);
+        getArticleById(article_id)
+            .then((data) => {
+                setArticle(data.article);
+                setIsLoading(false);
+            })
+            .catch(err => {
+                console.error("Error getting article:", err);
+                setError("Failed to load article");
+                setIsLoading(false);
+            });
+        }, [article_id]);
```

Comment:
You wouldn't need to have the article_id in here as the id can't change without the whole entire page being reloaded. You still need the array for sure but it doesn't need an element.